certificates/src/lib.rs

```diff
@@ -21,30 +23,60 @@ pub enum CertificateError {
     CertificateExpired,
 
     #[error("Failed to parse private key")]
-    BadPrivateKey(openssl::error::ErrorStack),
+    BadPrivateKey(#[source] openssl::error::ErrorStack),
 
     #[error("Certificate and private key do not match")]
     Mismatch,
 
-    #[error("Unexpected error: {0}")]
-    Unexpected(openssl::error::ErrorStack),
+    #[error("Failed to convert common name to UTF8")]
+    NonUtf8CommonName(#[source] openssl::error::ErrorStack),
+
+    #[error("Certificate subject does not contain a common name")]
+    NoCommonName,
```

Comment:
```suggestion
    #[error("Certificate subject contains neither alternate names nor a common name")]
    NoName,
```

(feel free to ignore)